fbw-a380x/src/systems/systems-host/systems/LegacyFuel.ts

```diff
@@ -0,0 +1,431 @@
+
+  private readonly triggerStates = new Map<number, ConsumerSubject<boolean>>();
+
+  private readonly junctionSettings = new Map<number, ConsumerSubject<number>>();
+
+  private refuelInProgress = false;
+
+  constructor(private readonly bus: EventBus) {
+    KeyEventManager.getManager(bus).then((manager) => {
+      this.keyEventManager = manager;
+    });
+
+    for (let index = 1; index <= this.NUMBER_OF_TRIGGERS; index++) {
+      const element = ConsumerSubject.create(this.sub.on(`fuel_trigger_status_${index}`), false);
+      this.triggerStates.set(index, element);
+    }
+
+    for (let index = 1; index <= this.NUMBER_OF_JUNCTIONS; index++) {
+      const element = ConsumerSubject.create(this.sub.on(`fuel_junction_setting_${index}`), 1);
+      this.junctionSettings.set(index, element);
+    }
+  }
+
+  init() {
+    const fuelWeight = SimVar.GetSimVarValue('FUEL TOTAL QUANTITY WEIGHT', 'kilograms');
+    SimVar.SetSimVarValue('L:A32NX_FUEL_DESIRED', 'kilograms', fuelWeight);
+    Wait.awaitSubscribable(GameStateProvider.get(), (state) => state === GameState.ingame, true).then(() => {
+      this.checkEmptyTriggers();
+    });
+  }
+
+  private checkEmptyTriggers(): void {
+    if (
+      (this.leftInnerTankQty.get() < 0.1 && !this.triggerStates.get(11).get()) ||
+      (this.leftInnerTankQty.get() >= 1 && this.triggerStates.get(11).get())
+    ) {
+      this.toggleTrigger(11);
+    }
+
+    if (
+      (this.rightInnerTankQty.get() < 0.1 && !this.triggerStates.get(12).get()) ||
+      (this.rightInnerTankQty.get() >= 1 && this.triggerStates.get(12).get())
+    ) {
+      this.toggleTrigger(12);
+    }
+
+    if (
+      (this.leftMidTankQty.get() < 0.1 && !this.triggerStates.get(22).get()) ||
+      (this.leftMidTankQty.get() >= 1 && this.triggerStates.get(22).get())
+    ) {
+      this.toggleTrigger(22);
+    }
+
+    if (
+      (this.rightMidTankQty.get() < 0.1 && !this.triggerStates.get(23).get()) ||
+      (this.rightMidTankQty.get() >= 1 && this.triggerStates.get(23).get())
+    ) {
+      this.toggleTrigger(23);
+    }
+
+    if (
+      this.rightInnerTankQty.get() < 0.1 &&
+      this.leftInnerTankQty.get() < 0.1 &&
+      this.rightMidTankQty.get() < 0.1 &&
+      this.leftMidTankQty.get() < 0.1
+    ) {
+      // both mid and inner tanks are empty
+
+      this.setJunctionOption(10, 3);
+    } else if (this.rightInnerTankQty.get() >= 1 || this.leftInnerTankQty.get() >= 1) {
+      // inner tanks arent empty
+
+      this.setJunctionOption(10, 1);
+    } else {
+      // mid tanks arent empty but inner tanks are
+
+      this.setJunctionOption(10, 2);
+    }
+
+    if (
+      (this.trimTankQty.get() < 0.1 && !this.triggerStates.get(32).get()) ||
+      (this.trimTankQty.get() >= 1 && this.triggerStates.get(32).get())
+    ) {
+      this.toggleTrigger(32);
+    }
+  }
+
+  public onUpdate(): void {
+    const onGround = SimVar.GetSimVarValue('SIM ON GROUND', 'bool');
```

Comment:
Might consider moving this to a publisher as well(SIM_ON_GROUND is also used elsewhere on systems-host), but maybe not for this PR

Reply:
Even more nitpick, anything using SIM ON GROUND should really be in extras-host because it's not an aircraft system (they use other aircraft systems to determine on ground/in-flight status).

Reply:
I was kinda going back and forth on that and ended up just using the existing file. In the end this code isnt designed to stay around for long (hopefully) and be replaced by our own fqms.